worlds/rabi_ribi/existing_randomizer/dataparser.py

```diff
@@ -215,35 +249,78 @@ def define_default_expressions(variable_names_set):
         "SPEED5": "SPEED_BOOST_LV3 & SPEEDY",
     })
     def1.update(def3)
+
+    expr = lambda s : parse_expression(s, variable_names_set, def1)
+    def4 = expr_all({
+        "1TILE_ZIP": "SLIDE_ZIP",
+        "2TILE_ZIP": "SLIDE_ZIP & ADV_VHARD",
+        "3TILE_ZIP": "SLIDE_ZIP & HARD",
+        "4TILE_ZIP": "SLIDE_ZIP & HARD",
+        "5TILE_ZIP": "RABI_SLIPPERS & SLIDE_ZIP & ADV_VHARD",
+
+        "5TILE_WALL_CLIMB": "\
+            AIR_JUMP | AIR_DASH | (ADV_VHARD & RABI_SLIPPERS & AMULET_FOOD)\
+            | (ADV_EXT & WALL_JUMP & 2_AMULET_FOOD & (BUNNY_AMULET | STUPID))\
+            | (OBS_STUPID & BORING & 6_AMULET_FOOD)\
+        ",
+
+        "5TILE_WALL_CLIMB_BUNSTRIKE": "\
+            (RABI_SLIPPERS & SLIDE_JUMP_BUNSTRIKE)\
+            | (ADV_EXT & SLIDE_JUMP_BUNSTRIKE_CANCEL & 2_AMULET_FOOD)\
+        ",
+    })
+    def1.update(def4)
+
     return def1
 
 def shufflable_gift_item_map_modifications():
+    # AP Change: Use os.path.join
     return [
         os.path.join('existing_randomizer', 'maptemplates', 'shuffle_gift_items', 'mod_p_hairpin.txt'),
         os.path.join('existing_randomizer', 'maptemplates', 'shuffle_gift_items', 'mod_bunstrike_speedboost.txt')
     ]
 
-def get_default_areaids():
-    return list(range(10))
+# AP Change: Moved get_default_areaids to utility.py
 
-TOWN_MEMBERS = ('TM_COCOA','TM_ASHURI','TM_RITA','TM_CICINI','TM_SAYA','TM_SYARO','TM_PANDORA','TM_NIEVE','TM_NIXIE','TM_ARURAUNE','TM_SEANA','TM_LILITH','TM_VANILLA','TM_CHOCOLATE','TM_KOTRI','TM_KEKE_BUNNY',)
+TOWN_MEMBERS = (
+    'TM_COCOA', 'TM_ASHURI', 'TM_RITA', 'TM_CICINI',
+    'TM_SAYA', 'TM_SYARO', 'TM_PANDORA', 'TM_NIEVE',
+    'TM_NIXIE', 'TM_ARURAUNE', 'TM_SEANA', 'TM_LILITH',
+    'TM_VANILLA', 'TM_CHOCOLATE', 'TM_KOTRI', 'TM_KEKE_BUNNY',
+    )
 def count_town_members(variables):
     return sum(1 for tm in TOWN_MEMBERS if variables[tm])
 
+# removed TM_VANILLA, TM_CHOCOLATE, TM_CICINI, TM_SYARO, TM_NIEVE, and TM_NIXIE from requirements since TM_SEANA is encompasses them
```

Comment:
This code only counts Seana as one town member no? Would recommend just listing them all anyways for readability purposes (easier to parse the underlying logic in one's head)

Reply:
Nit: could just call count_town_members here.

Reply:
I agree, but this is from the standalone randomizer